internal/service/domain/sensor/sensor.go

```diff
@@ -100,6 +106,46 @@ func (s *SensorService) Delete(ctx context.Context, id string) error {
 	return nil
 }
 
+func (s *SensorService) RunStatusUpdater(ctx context.Context, interval time.Duration) {
```

Comment:
I think I would move this whole function into a separate component that is called and created within the service. That way you encapsulate the function a bit, which is a bit nicer